orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/config/CloudDriverConfiguration.java

```diff
@@ -129,43 +122,94 @@ private <T> SelectableService buildReadOnlyService(Class<T> type) {
-                ServiceSelector selector =
-                    new DefaultServiceSelector(
-                        buildService(type, url.getBaseUrl()), url.getPriority(), url.getConfig());
-
-                if (url.getConfig() != null && url.getConfig().get("selectorClass") != null) {
-                  try {
-                    Class<ServiceSelector> selectorClass =
-                        (Class<ServiceSelector>)
-                            Class.forName(url.getConfig().get("selectorClass").toString());
-                    selector =
-                        (ServiceSelector)
-                            selectorClass.getConstructors()[0].newInstance(
-                                selector.getService(), selector.getPriority(), url.getConfig());
-                  } catch (Exception e) {
-                    log.error(
-                        "Failed to create selector for class {}",
-                        url.getConfig().get("selectorClass"));
-
-                    throw new RuntimeException(e);
-                  }
-                }
-
-                selectors.add(selector);
-              });
+      List<ServiceSelector> selectors =
+          getServiceSelectors(
+              cloudDriverConfigurationProperties.getCloudDriverReadOnlyBaseUrls(), type);
+      return new SelectableService(selectors);
+    }
+
+    /**
+     * Builds a {@link SelectableService} for a write Clouddriver endpoint. This method gets any
+     * writeonly URLs configured in the profile and then for each constructs a {@link
+     * ServiceSelector} based on the criteria configured.
+     *
+     * @param type the class to construct
+     * @param <T> the type of class to construct
+     * @return a {@link SelectableService} configured with the {@link ServiceSelector} instances
+     *     from the profile
+     */
+    public <T> SelectableService buildWriteableService(Class<T> type) {
+      List<String> urls =
+          cloudDriverConfigurationProperties.getCloudDriverWriteOnlyBaseUrls().stream()
+              .map(CloudDriverConfigurationProperties.BaseUrl::getBaseUrl)
+              .collect(Collectors.toList());
+
+      if (urls.isEmpty()
+          || urls.stream()
+              .allMatch(
+                  url -> url.equals(cloudDriverConfigurationProperties.getCloudDriverBaseUrl()))) {
+        log.info(
+            "writeonly URL not configured for clouddriver, using default clouddriver {} for {}",
+            cloudDriverConfigurationProperties.getCloudDriverBaseUrl(),
+            type.getSimpleName());
+      }
 
+      List<ServiceSelector> selectors =
+          getServiceSelectors(
+              cloudDriverConfigurationProperties.getCloudDriverWriteOnlyBaseUrls(), type);
       return new SelectableService(selectors);
     }
+
+    private <T> List<ServiceSelector> getServiceSelectors(List<BaseUrl> baseUrls, Class<T> type) {
+      List<ServiceSelector> serviceSelectors = new ArrayList<>();
+
+      baseUrls.forEach(
```

Comment:
I did think about that, but Gate also does the same thing (with a different config structure because.. reasons..). Agree it would be nice to shift this stuff out of here.